### DISCUSSION TITLE
How to integrate the Blazor webassembly in WPF (webview2)

### DISCUSSION THREAD

**Discussion by venkakat83** [MicrosoftEdge/WebView2Feedback#3886] [Q&A]
Currently we have web application running on Angular. And we would like to integrate angular based web component into existing windows based desktop application.

Our main functionality:-

Angular Component and Desktop app should communicate each other.
Can some one provide some insights to work on? will this can be achieved with integrating the Blazor webassmebly into webview2 ? If yes, please guide and kindly share if we have any demo project.

**Comment by champnic** [comment]
Your scenario sounds achievable with WebView2, using scripting or host objects or web messaging for your native <> web interop.
**Samples:** https://github.com/MicrosoftEdge/WebView2Samples
**Getting Started (WPF):** https://docs.microsoft.com/en-us/microsoft-edge/webview2/get-started/wpf

I'm not very familiar with Blazor - is there a reason you are looking at it? @Eilon might be able to help answer Blazor questions you might have.

**Comment by venkakat83** [comment]
@champnic  Thank you for your replay. 
 
Integration of Web-components  into Blazor webassembly and hosting in webview is easier than hosting directly in webview2
